phalcon/mvc/model/manager.zep

```diff
@@ -1805,4 +1811,855 @@ class Manager implements ManagerInterface, InjectionAwareInterface, EventsAwareI
+	{
+		var distinctColumn, groupColumn, columns,
+			bindParams, bindTypes, resultset, cache, firstRow, groupColumns,
+			builder, query;
+
+		if !fetch groupColumn, parameters["column"] {
+			let groupColumn = "*";
+		}
+
+		/**
+		 * Builds the columns to query according to the received parameters
+		 */
+		if fetch distinctColumn, parameters["distinct"] {
+			let columns = functionName . "(DISTINCT " . distinctColumn . ") AS " . alias;
+		} else {
+			if fetch groupColumns, parameters["group"] {
+				let columns = groupColumns . ", " . functionName . "(" . groupColumn . ") AS " . alias;
+			} else {
+				let columns = functionName . "(" . groupColumn . ") AS " . alias;
+			}
+		}
+
+		/**
+		 * Builds a query with the passed parameters
+		 */
+		let builder = this->createBuilder(parameters);
+		builder->columns(columns);
+		builder->from(modelClass);
+
+		let query = builder->getQuery();
+
+		/**
+		 * Check for bind parameters
+		 */
+		let bindParams = null, bindTypes = null;
+		if fetch bindParams, parameters["bind"] {
+			fetch bindTypes, parameters["bindTypes"];
+		}
+
+		/**
+		 * Pass the cache options to the query
+		 */
+		if fetch cache, parameters["cache"] {
+			query->cache(cache);
+		}
+
+		/**
+		 * Execute the query
+		 */
+		let resultset = query->execute(bindParams, bindTypes);
+
+		/**
+		 * Return the full resultset if the query is grouped
+		 */
+		if isset parameters["group"] {
+			return resultset;
+		}
+
+		/**
+		 * Return only the value in the first result
+		 */
+		let firstRow = resultset->getFirst();
+		return firstRow->{alias};
+	}
+
+	/**
+	 * Inserts or updates a model instance. Returning true on success or false otherwise.
+	 *
+	 *<code>
+	 * // Creating a new robot
+	 * $robot = new Robots();
+	 *
+	 * $robot->type = "mechanical";
+	 * $robot->name = "Astro Boy";
+	 * $robot->year = 1952;
+	 *
+	 * $modelsManager->save($robot);
+	 *
+	 * // Updating a robot name
+	 * $robot = $modelsManager->findFirst(
+	 *     Robots::class,
+	 *     [
+	 *         "id = 100"
+	 *     ]
+	 * );
+	 *
+	 * $robot->name = "Biomass";
+	 *
+	 * $modelsManager->save($robot);
+	 *</code>
+	 *
+	 * @param array data
+	 * @param array whiteList
+	 * @return boolean
+	 */
+	public function save(<ModelInterface> model, var data = null, var whiteList = null) -> boolean
+	{
+		var metaData, related, schema, writeConnection, readConnection,
+			source, table, identityField, exists, success;
+
+		let metaData = model->getModelsMetaData();
+
+		if typeof data == "array" && count(data) > 0 {
+			model->assign(data, null, whiteList);
+		}
+
+		/**
+		 * Create/Get the current database connection
+		 */
+		let writeConnection = model->getWriteConnection();
+
+		/**
+		 * Fire the start event
+		 */
+		model->fireEvent("prepareSave");
+
+		/**
+		 * Save related records in belongsTo relationships
+		 */
+		let related = model->_related;
+		if typeof related == "array" {
+			if model->_preSaveRelatedRecords(writeConnection, related) === false {
```

Comment:
There should be something like `Phalcon\Mvc\Model\Worker` or `Phalcon\Mvc\Model\UnitOfWork` which will do this.